sdks/java/core/src/main/java/org/apache/beam/sdk/util/state/InMemoryTimerInternals.java

```diff
@@ -97,6 +98,18 @@ public Instant getNextTimer(TimeDomain domain) {
   }
 
   @Override
+  public void setTimer(StateNamespace namespace, String timerId, Instant target,
+      TimeDomain timeDomain) {
+    WindowTracing.trace(
+        "TestTimerInternals.setTimer: {}/{} {} {}", namespace, timerId, target, timeDomain);
+    // TODO: actually track timers by ID
+    TimerData timer = TimerData.of(namespace, target, timeDomain);
+    if (existingTimers.add(timer)) {
```

Comment:
Mostly out of intent to delete that method, and because you cannot implement the ID-based method in terms of the `TimerData`-based method, if it were implemented.

Per Thomas's comment above, I'm replacing them all with unsupported exceptions.